go/tasks/pluginmachinery/catalog/client.go

```diff
@@ -73,10 +75,55 @@ func NewCatalogEntry(outputs io.OutputReader, status Status) Entry {
 	return Entry{outputs: outputs, status: status}
 }
 
+// Indicates the Reservation in Catalog
+type ReservationEntry struct {
```

Comment:
Maybe add a few unit tests for this, admittedly trivial, file?

While we are at it, can we add docs on public types/funcs/methods?

Reply:
done